pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/RealtimeTableDataManager.java

```diff
@@ -658,11 +656,10 @@ private void downloadSegmentFromPeer(String segmentName, String downloadScheme,
     try {
       tempRootDir = getTmpSegmentDataDir("tmp-" + segmentName + "." + System.currentTimeMillis());
       File segmentTarFile = new File(tempRootDir, segmentName + TarGzCompressionUtils.TAR_GZ_FILE_EXTENSION);
-      // First find servers hosting the segment in a ONLINE state.
-      List<URI> peerSegmentURIs = PeerServerSegmentFinder.getPeerServerURIs(segmentName, downloadScheme, _helixManager);
       // Next download the segment from a randomly chosen server using configured scheme.
-      SegmentFetcherFactory.getSegmentFetcher(downloadScheme).fetchSegmentToLocal(peerSegmentURIs, segmentTarFile);
-      _logger.info("Fetched segment {} from: {} to: {} of size: {}", segmentName, peerSegmentURIs, segmentTarFile,
+      SegmentFetcherFactory.getSegmentFetcher(downloadScheme)
+          .fetchSegmentToLocal(segmentName, segmentTarFile, _helixManager, downloadScheme);
```

Comment:
Do you think it's a good idea to expose helixManager in SegmentFetcher logic? Can we keep the logic of updating peerSegmentURIs with retries here itself?  
Segment fetcher should just fetch segment given a URI.

Reply:
HelixManager represents the Pinot cluster controller and is used for data/segment discovery. So from this view, I am not too concerned about passing it to segment fetcher -- which is then use it for retryable segment discovery.